drape_frontend/rule_drawer.cpp

```diff
@@ -289,105 +286,109 @@ void RuleDrawer::ProcessAreaStyle(FeatureType & f, Stylist const & s, TInsertSha
     areaMinHeight = static_cast<float>((rectMercator.SizeX() + rectMercator.SizeY()) * 0.5);
   }
 
-  bool applyPointStyle = s.m_pointStyleExists;
+  bool applyPointStyle = s.m_symbolRule != nullptr || s.m_captionRule != nullptr ||
+                         s.m_houseNumberRule != nullptr;
   if (applyPointStyle)
   {
     if (!is3dBuilding)
     {
       // Loads geometry of the feature.
-      featureCenter = feature::GetCenter(f, zoomLevel);
+      featureCenter = feature::GetCenter(f, m_zoomLevel);
     }
     applyPointStyle = m_globalRect.IsPointInside(featureCenter);
   }
 
-  ApplyAreaFeature apply(m_context->GetTileKey(), insertShape, f.GetID(),
+  double areaDepth = NAN;
```

Comment:
Aha, here it is :) I'd prefer simple -1 value and check that areaDepth > 0 in a Priprity function.

Reply:
areaDepth should be in the `[-11000,-10100)` range, but we might move/resize the ranges, so using NAN is more reliable.

Reply:
Please do not use NAN. We use [-ffast-math](https://stackoverflow.com/a/22135559) in our code.

Reply:
Changed to an out-of-range const.
PTAL!